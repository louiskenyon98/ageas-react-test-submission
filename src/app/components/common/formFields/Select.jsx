import React from 'react';
import PropTypes from 'prop-types';

class Select extends React.PureComponent {
    constructor(props) {
        super(props);
        this.onChange = this.onChange.bind(this);
    }

    onChange(e) {
        this.props.onChange(e.target.value);
    }

    returnOptions() {
        const {options} = this.props;
        if(Array.isArray(options) && options.length) {
            return options.map(e => {
                return (
                    <option key={e.value} value={e.value}>{e.label}</option>
                );
            });
        }
    }

    render() {
        return (
            <select value={this.props.value} onChange={this.onChange}>
                {this.returnOptions()}
            </select>
        )
    }
}

Select.propTypes = {
    options: PropTypes.array,
    onChange: PropTypes.func,
    value: PropTypes.string
};

export default Select;
