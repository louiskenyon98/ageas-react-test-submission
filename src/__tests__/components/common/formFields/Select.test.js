import React from 'react';
import {shallow} from 'enzyme';
import Select from '../../../../app/components/common/formFields/Select';

describe('Select', () => {
    let props = {};
    let wrapper = () => shallow(<Select {...props}/>);

    beforeEach(() => {
        props = {
            options: [{
                value: 'hello',
                label: 'Hello World'
            },{
                value: 'foo',
                label: 'Foo Bar'
            },{
                value: 'mike',
                label: 'Big Mike'
            }],
            value: 'foo',
            onChange: jest.fn()
        }
    });

    describe('render', () => {
        describe('options', () => {
            it('should render a select component with multiple options', () => {
                expect(wrapper()).toMatchSnapshot();
                // console.log(wrapper().debug());
            });

            it('should return no options when no options are passed in', () => {
                props.options = undefined;
                expect(wrapper()).toMatchSnapshot();
            });
        });
    });

    describe('functionality', () => {
        it('should call onChange prop onChange of select', () => {
            expect(props.onChange).not.toHaveBeenCalled();
            expect(props.onChange).toHaveBeenCalledTimes(0);
            wrapper().find('select').first().props().onChange({
                target: {
                    value: 'mike'
                }
            });
            expect(props.onChange).toHaveBeenCalledTimes(1);
            expect(props.onChange).toHaveBeenCalledWith('mike');
        });
    });
});
