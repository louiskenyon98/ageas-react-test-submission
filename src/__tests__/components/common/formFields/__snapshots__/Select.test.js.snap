// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select render options should render a select component with multiple options 1`] = `
<select
  onChange={[Function]}
  value="foo"
>
  <option
    key="hello"
    value="hello"
  >
    Hello World
  </option>
  <option
    key="foo"
    value="foo"
  >
    Foo Bar
  </option>
  <option
    key="mike"
    value="mike"
  >
    Big Mike
  </option>
</select>
`;

exports[`Select render options should return no options when no options are passed in 1`] = `
<select
  onChange={[Function]}
  value="foo"
/>
`;

exports[`Select should render a select component with multiple options 1`] = `
<select
  onChange={[Function]}
  value="foo"
>
  <option
    key="hello"
    value="hello"
  >
    Hello World
  </option>
  <option
    key="foo"
    value="foo"
  >
    Foo Bar
  </option>
  <option
    key="mike"
    value="mike"
  >
    Big Mike
  </option>
</select>
`;

exports[`Select should return no options when no options are passed in 1`] = `
<select
  onChange={[Function]}
  value="foo"
/>
`;
